pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/json/ImmutableJsonIndexReader.java

```diff
@@ -76,6 +77,63 @@ public ImmutableJsonIndexReader(PinotDataBuffer dataBuffer, int numDocs) {
     _docIdMapping = dataBuffer.view(invertedIndexEndOffset, docIdMappingEndOffset, ByteOrder.LITTLE_ENDIAN);
   }
 
+  /**
+   * Accepts a JSON key and array of docIds used to filter the response
+   * return a String[] where String[i] gives the value of $.key for document i
+   */
+  @Override
+  public String[] getValuesForKeyAndDocs(String key, int[] docIds) {
+    ImmutableRoaringBitmap docIdMask = ImmutableRoaringBitmap.bitmapOf(docIds);
+    int[] dictIds = getDictIdsForKey(key);
+    String[] values = new String[(int) _numDocs];
+    for (int dictId = dictIds[0]; dictId < dictIds[1]; dictId++) {
+      // get docIds from posting list, convert these to the actual docIds
+      ImmutableRoaringBitmap flattenedDocIds = _invertedIndex.getDocIds(dictId);
+      PeekableIntIterator it = flattenedDocIds.getIntIterator();
+      MutableRoaringBitmap postingList = new MutableRoaringBitmap();
+      while (it.hasNext()) {
+        postingList.add(getDocId(it.next()));
+      }
+      // if posting list does not contain relevant docIds, skip the dictionary lookup
+      postingList.and(docIdMask);
+      if (postingList.isEmpty()) {
+        continue;
+      }
+
+      // dictionary value lookup, stripping the path prefix
+      String val = _dictionary.getStringValue(dictId).substring(key.length() + 1);
+
+      // add value to padded array
+      for (int docId : postingList) {
+        values[docId] = val;
```

Comment:
This won't work if the same doc contains multiple flattened docs of the same key but different value. They will override each other. We need to document this limitation.

Reply:
Is it okay to document this in the function reference documentation? 

Looking at JSON standards there doesn't seem to be a standard behavior for handling duplicate keys